src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -164,6 +166,113 @@ public static FormattableString CreateDefaultPropertyDescription(string nameToUs
 
         public SourcePropertySerializationMapping? SerializationMapping { get; }
 
+        /// <summary>
+        /// This method attempts to retrieve the description for each of the union type items
+        /// by retrieving the friendly nam of the type. For items that are lists,
+        /// the description will include details about the element type.
+        /// If the friendly name cannot be retrieved, then the name of the type is used to construct the description.
+        /// </summary>
+        /// <param name="unionItems">the list of union type items.</param>
+        /// <returns>A list of strings representing the description of each union item and their corresponding
+        /// CSharp type frinedly name.
+        /// </returns>
+        public static IReadOnlyList<string> GetUnionTypesDescriptions(IList<CSharpType> unionItems)
+        {
+            var values = new List<string>();
+
+            foreach (CSharpType item in unionItems)
+            {
+                if (item != null)
+                {
+                    var friendlyTypeName = item.TryGetCSharpFriendlyName(out var keywordName) ? keywordName : item.Name;
+
+                    if (!friendlyTypeName.IsNullOrEmpty())
+                    {
+                        var description = $"<description><see cref=\"{friendlyTypeName}\"/></description>";
```

Comment:
If you use the CSharpType here as in the input the namespaces get added automatically by the writer.

```c#
$"<description><see cref=\"{item}\"/></description>"
```